File: src/main/java/com/atsuishio/superbwarfare/data/gun/ProjectileInfo.java
package com.atsuishio.superbwarfare.data.gun;

import com.atsuishio.superbwarfare.data.DeserializeFromString;
import com.atsuishio.superbwarfare.data.IDBasedData;
import com.google.gson.JsonObject;
import com.google.gson.annotations.SerializedName;

public class ProjectileInfo implements IDBasedData<ProjectileInfo>, DeserializeFromString {

    @SerializedName("Type")
    public String type = "superbwarfare:projectile";

    @SerializedName("Data")
    public JsonObject data;

    @Override
    public String getId() {
        return type;
    }

    @Override
    public void setId(String id) {
        this.type = id;
    }

    @Override
    public void deserializeFromString(String str) {
        this.type = str;
    }
}

File: src/main/java/com/atsuishio/superbwarfare/data/gun/ProjectileInfo.kt
package com.atsuishio.superbwarfare.data.gun

import com.atsuishio.superbwarfare.data.DeserializeFromString
import com.atsuishio.superbwarfare.data.IDBasedData
import com.atsuishio.superbwarfare.serialization.kserializer.SerializedGsonObject
import com.google.gson.annotations.SerializedName
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class ProjectileInfo : IDBasedData<ProjectileInfo>, DeserializeFromString {
    @SerializedName("Type")
    @SerialName("Type")
    var itemId: String = "superbwarfare:projectile"

    override fun getId() = itemId
    override fun setId(id: String) {
        this.itemId = id
    }

    @JvmField
    @SerializedName("Data")
    @SerialName("Data")
    var data: SerializedGsonObject? = null

    override fun deserializeFromString(str: String) {
        this.itemId = str
    }
}
